flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/OperationConverterUtils.java

```diff
@@ -144,6 +147,163 @@ public static Operation convertChangeColumn(
         // TODO: handle watermark and constraints
     }
 
+    public static Operation convertRenameColumn(
+            ObjectIdentifier tableIdentifier,
+            String originColumnName,
+            String newColumnName,
+            CatalogTable originTable,
+            ResolvedSchema originResolveSchema) {
+        Schema originSchema = originTable.getUnresolvedSchema();
```

Comment:
Can we reuse `SchemaResolver` to simplify this check just like ALTER TABLE ADD?
```java
Schema originSchema = originTable.getUnresolvedSchema();
        List<Schema.UnresolvedColumn> newColumns =
                originSchema.getColumns().stream()
                        .map(
                                column -> {
                                    if (column.getName().equals(originColumnName)) {
                                        return convertColumn(column, newColumnName);
                                    } else {
                                        return column;
                                    }
                                })
                        .collect(Collectors.toList());

        Schema updatedSchema =
                Schema.newBuilder().fromSchema(originSchema).replaceColumns(newColumns).build();
        // try {
        //                schemaResolver.resolve(updatedSchema);
        //                return updatedSchema;
        //            } catch (Exception e) {
        //                throw new ValidationException(
        //                        String.format("%s%s", EX_MSG_PREFIX, e.getMessage()), e);
        //            }
        return new AlterTableSchemaOperation(
                tableIdentifier,
                CatalogTable.of(
                        updatedSchema,
                        originTable.getComment(),
                        originTable.getPartitionKeys(),
                        originTable.getOptions()));
```

Reply:
The current implementation also modify the pk list if the original column is in the list.